components/places/src/storage/history.rs

```diff
@@ -2739,6 +2760,20 @@ mod tests {
         )
         .unwrap();
         assert!(maybe_row.is_none(), "Shouldn't insert overlong URL");
+
+        let maybe_row_preview = apply_observation(
+            &conn,
+            VisitObservation::new(Url::parse("https://www.example.com/").unwrap())
+                .with_preview_image_url(Some(url))
+                .with_visit_type(VisitTransition::Link)
+                .with_at(Timestamp::now()),
+        )
+        .unwrap();
+        assert!(
+            maybe_row_preview.is_none(),
+            "Shouldn't insert overlong preview URL"
```

Comment:
so yeah, I think this test would be that the visit was recorded but the url is None. And ideally another similar test that an invalid url throws.